pig/src/main/java/org/apache/iceberg/pig/IcebergPigInputFormat.java

```diff
@@ -243,6 +243,7 @@ private boolean advance() throws IOException {
       return true;
     }
 
+    @SuppressWarnings("ByteBufferBackingArray")
```

Comment:
I don't think this is correct. This calls `buffer.get(new byte[...])`, which returns the `ByteBuffer` that `get` was called on. It is no different than returning `new DataByteArray(buffer.array())`. Calling `get` shows that the intent was to read the bytes into a new array and pass that to create a `DataByteArray`.

The correct implementation is this:
```java
ByteBuffer buffer = (ByteBuffer) value;
byte[] bytes = new byte[buffer.remaining()];
buffer.get(bytes);
return new DataByteArray(bytes);
```

Reply:
Ok. Thanks for the insight. This was one admittedly that I suppressed but wasn't sure about. I'm in agreement though, due to the same arguments I've given elsewhere. I know much less about Pig, but I think it's better to be safe than sorry here if we don't know about the exact behavior of the input source, not or in the future. 

Should I update it in this PR or another PR?

Reply:
And do I need to restore the buffer position as well? In my implementation above that uses `remaining` above in GenericOrcWriters that we're also discussing, I returned the original buffer (`value` here) to its original position. IIUC, they'd be backed by the same data as the casted `buffer`.

Reply:
It's worth noting that the java doc of ByteBuffer states

```
     * Unrelated stuff about BufferUnderflowException....
     *
     * <p> Otherwise, this method copies <tt>length</tt> bytes from this
     * buffer into the given array, starting at the current position of this
     * buffer and at the given offset in the array.  The position of this
     * buffer is then incremented by <tt>length</tt>.
     * ....
```

Reply:
I updated this. I also moved the position of `value` (or `buffer` as it were) back to `remaining` before returning the copy. Is that necessary / will it lead to issues? I'm not sure what's needed in this case.

Reply:
We usually `duplicate` the buffer before calling `get` to avoid changing the original buffer's state.

And this can be updated in this PR, since the scope here is to fix the errors and suppress any other warnings.

Reply:
I see. That makes sense. Thanks for helping me to understand better. And you're right, fixing the issues is definitely in scope. That's why I started this endeavor 😅